storage/disk/disk_test.go

```diff
@@ -242,40 +262,49 @@ func TestTruncate(t *testing.T) {
 			}
 		}
 		`
-		jsn := util.MustUnmarshalJSON([]byte(expected))
+	jsn := util.MustUnmarshalJSON([]byte(expected))
 
-		if !reflect.DeepEqual(jsn, actual) {
-			t.Fatalf("Expected reader's read to be %v but got: %v", jsn, actual)
-		}
+	if !reflect.DeepEqual(jsn, actual) {
+		t.Fatalf("Expected reader's read to be %v but got: %v", jsn, actual)
+	}
 
-		s.Abort(ctx, txn)
+	s.Abort(ctx, txn)
 
-		txn = storage.NewTransactionOrDie(ctx, s)
-		ids, err := s.ListPolicies(ctx, txn)
-		if err != nil {
-			t.Fatal(err)
-		}
+	txn = storage.NewTransactionOrDie(ctx, s)
+	ids, err := s.ListPolicies(ctx, txn)
+	if err != nil {
+		t.Fatal(err)
+	}
 
-		expectedIds := map[string]struct{}{"/policy.rego": {}, "/roles/policy.rego": {}}
+	expectedIds := map[string]struct{}{"/policy.rego": {}, "/roles/policy.rego": {}}
 
-		for _, id := range ids {
-			if _, ok := expectedIds[id]; !ok {
-				t.Fatalf("Expected list policies to contain %v but got: %v", expectedIds, id)
-			}
+	for _, id := range ids {
+		if _, ok := expectedIds[id]; !ok {
+			t.Fatalf("Expected list policies to contain %v but got: %v", expectedIds, id)
 		}
+	}
 
-		bs, err := s.GetPolicy(ctx, txn, "/policy.rego")
-		expectedBytes := []byte("package foo\n p = 1")
-		if err != nil || !reflect.DeepEqual(expectedBytes, bs) {
-			t.Fatalf("Expected get policy to return %v but got: %v (err: %v)", expectedBytes, bs, err)
-		}
+	bs, err := s.GetPolicy(ctx, txn, "/policy.rego")
+	expectedBytes := []byte("package foo\n p = 1")
+	if err != nil || !reflect.DeepEqual(expectedBytes, bs) {
+		t.Fatalf("Expected get policy to return %v but got: %v (err: %v)", expectedBytes, bs, err)
+	}
 
-		bs, err = s.GetPolicy(ctx, txn, "/roles/policy.rego")
-		expectedBytes = []byte("package bar\n p = 1")
-		if err != nil || !reflect.DeepEqual(expectedBytes, bs) {
-			t.Fatalf("Expected get policy to return %v but got: %v (err: %v)", expectedBytes, bs, err)
-		}
-	})
+	bs, err = s.GetPolicy(ctx, txn, "/roles/policy.rego")
+	expectedBytes = []byte("package bar\n p = 1")
+	if err != nil || !reflect.DeepEqual(expectedBytes, bs) {
+		t.Fatalf("Expected get policy to return %v but got: %v (err: %v)", expectedBytes, bs, err)
+	}
+
+	// Close and re-open store
```

Comment:
This is the new thing -- it would catch the previous failure, which would yield to not being able to deref the symlink `foobar/foobar/backupXXXX`.